src/net_processing.cpp

```diff
@@ -2394,10 +2412,20 @@ bool static ProcessMessage(CNode* pfrom, const std::string& strCommand, CDataStr
                 // us sync -- disconnect if using an outbound slot (unless
                 // whitelisted or addnode).
                 if (!(pfrom->fInbound || pfrom->fWhitelisted || pfrom->m_manual_connection)) {
+                    LogPrintf("Disconnecting outbound peer %d -- headers chain has insufficient work\n", pfrom->GetId());
                     pfrom->fDisconnect = true;
                 }
             }
         }
+
+        if (!pfrom->fDisconnect && !pfrom->fInbound && !pfrom->fOneShot && !pfrom->fFeeler && !pfrom->m_manual_connection) {
```

Comment:
this condition seems to repeat twice in this commit (apart from `!pfrom->fDisconnect`), probably best to have this in its own function